### DISCUSSION TITLE
Is there like hard-reload/redirect?

### DISCUSSION THREAD

**Discussion by hibangun** [remix-run/remix#2183] [Q&A] [answered]
I have `loader` and `action` in the same file and using mikro orm
The issue is after user submit the form, we redirect to the current page
```
export const action: ActionFunction = async ({ request }) => {
  const em = orm.em.fork()
  const url = new URL(request.url)

  ...
  return redirect(url.search || url.pathname)
}
```
 but the data is not updated until I manually refresh the page in browser

Is there like hard-reload/redirect? or double reload page?

**Comment by sergiodxa** [comment, marked as answer]
You can add reloadDocument to both Link and Form and they will do the navigation without JS, so they will do the hard-reload/redirect.

**Comment by kiliman** [comment]
When you say the data is not updated, what do you mean? Since you're redirecting to the same page, React won't re-render any components if props/state don't change.

For example, if you want to clear the form, you need to give it something that changes, since typically your inputs would use defaultValue prop (only used on initial render). Basically do something like this:
```ts
const location = useLocation()
<Form key={location.key}>....</Form>
```
This will cause the Form to be re-rendered after every navigation, even redirects to the same route.
